From c7a10878501dad7c348445365b2ad1336a01d7d8 Mon Sep 17 00:00:00 2001
From: Jayesh Choudhary <j-choudhary@ti.com>
Date: Thu, 12 Oct 2023 14:00:38 +0530
Subject: [PATCH 053/322] Revert "drm/tidss: Make zpos immutable for primary
 plane"

This reverts commit e0757b737833c0baee39cf54de96abb15a7b4a24.

Fixing zpos to 0 for two primary planes causes issue when one
video pipeline tries to use all other planes as overlay as we
cannot change the zpos higher than background.
This effectively means that we can use only two instead of three
planes as overlay (for J7 devices with 4 OVR and VP).
Therefore reverting the patch.

Signed-off-by: Jayesh Choudhary <j-choudhary@ti.com>
Reviewed-by: Aradhya Bhatia <a-bhatia1@ti.com>
---
 drivers/gpu/drm/tidss/tidss_plane.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/tidss/tidss_plane.c b/drivers/gpu/drm/tidss/tidss_plane.c
index f3e3de71f69b..5328d8b8cff0 100644
--- a/drivers/gpu/drm/tidss/tidss_plane.c
+++ b/drivers/gpu/drm/tidss/tidss_plane.c
@@ -211,11 +211,8 @@ struct tidss_plane *tidss_plane_create(struct tidss_device *tidss,
 
 	drm_plane_helper_add(&tplane->plane, &tidss_plane_helper_funcs);
 
-	if (type == DRM_PLANE_TYPE_PRIMARY)
-		drm_plane_create_zpos_immutable_property(&tplane->plane, 0);
-	else
-		drm_plane_create_zpos_property(&tplane->plane, tidss->num_planes, 1,
-					       num_planes - 1);
+	drm_plane_create_zpos_property(&tplane->plane, tidss->num_planes, 0,
+				       num_planes - 1);
 
 	ret = drm_plane_create_color_properties(&tplane->plane,
 						color_encodings,
-- 
2.43.0

